pkg/ast/structs.go

```diff
@@ -80,6 +80,17 @@ type ParseError struct {
 	Expected []string `json:"expected"`
 }
 
+type RelativeTimeModifier struct {
+	RelativeTimeOffset RelativeTimeOffset
+	Snap               string
+	AbsoluteTime       string
```

Comment:
Maybe rename the struct to TimeModifier because it has both Absolute and RelativeTimeModifier variables. And update it in a way that it is easy to differentiate between the two. Maybe like this 

```
type TimeModifier struct  {
	RelativeTime       RelativeTimeModifier
	AbsoluteTime       string
}

type RelativeTimeModifier struct {
	RelativeTimeOffset RelativeTimeOffset
	Snap               string
}
```

Update the variable names in the PEG file accordingly

Reply:
The issue is in future it could be the case that we need to support absolute time with relative like now-5d etc. That time it would be a relative time modifier but along with absolute time and relative time as well. Moreover, name TimeModifier is already taken for a struct so this was the only plausible name I could think of.

Reply:
I do not understand what you mean by this `we need to support absolute time with relative like now-5d`?

In that case, wouldn't it be a relative case? 

Also, for the other struct that is using the type `TimeModifier` , I think you are referring to `TimeModifiers` struct in `structs.go`. 

```
// Node is a node in the query parse tree
type Node struct {
	NodeType      NodeType
	Comparison    Comparison
	Left          *Node
	Right         *Node
	TimeModifiers *TimeModifiers
}

// TimeModifiers is a struct that holds the time modifiers for a query
type TimeModifiers struct {
	StartEpoch uint64
	EndEpoch   uint64
}
```

And based on the struct and it's use case, I would say refactor/rename this struct to `TimeRangeModifier`

Reply:
addressed in the new commit.